internal/services/tesla_api_service.go

```diff
@@ -0,0 +1,130 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/DIMO-Network/devices-api/internal/config"
+	"github.com/rs/zerolog"
+	"golang.org/x/oauth2"
+)
+
+//go:generate mockgen -source tesla_api_service.go -destination mocks/tesla_api_service_mock.go
+type TeslaAPIService interface {
+	CompleteTeslaAuthCodeExchange(authCode, redirectURI, region string) (*TeslaAuthCodeResponse, error)
+	GetVehicles(token, region string) ([]TeslaVehicle, error)
+}
+
+var teslaScopes = []string{"openid offline_access", "user_data", "vehicle_device_data", "vehicle_cmds", "vehicle_charging_cmds", "energy_device_data", "energy_device_data", "energy_cmds"}
+
+type GetVehiclesResponse struct {
+	Response []TeslaVehicle `json:"response"`
+}
+
+type TeslaError struct {
+	Error            string `json:"error"`
+	ErrorDescription string `json:"error_description"`
+	ReferenceID      string `json:"ReferenceID"`
+}
+
+type TeslaAuthCodeResponse struct {
+	AccessToken  string `json:"access_token"`
+	RefreshToken string `json:"refresh_token"`
+	IDToken      string `json:"id_token"`
+	ExpiresIn    int    `json:"expires_in"`
+	State        string `json:"state"`
+	TokenType    string `json:"token_type"`
```

Comment:
Are we using these two?

Reply:
Just included them for completeness since the API returns them, but I will remove.

Reply:
But you're not populating them, are you? If you were populating them it would be one thing.

Reply:
I'm populating tokenType, https://github.com/DIMO-Network/devices-api/pull/264/files/549f26e83f4b6488cd3a65ca0f88e54b0658b6fa#diff-e013fbe8446d2ccf3a214dda0a4f112e0e4850c595ddb2f43197e0d613a14973R83

The library doesn't seem to provide state